### PR TITLE
Feature layout

### PR DESCRIPTION
**Pull request by danielaSantaCruz**
* Screenshots to reports
* Partial table scenario 
* Layout implementation in legal entity feature

### CODE REVIEW

File: src/main/java/salesforce/entities/LegalEntity.java [hunk 1]
```diff
@@ -551,4 +551,20 @@ public String getTodayDate() {
         SimpleDateFormat dateFormat = new SimpleDateFormat("dd/M/yyyy HH:mm");
         return dateFormat.format(actualDate);
     }
+
+    /**
+     * Sets the values of the entity.
+     * @param legalEntity to be set.
```
Comment (Raymundogs) [inline src/main/java/salesforce/entities/LegalEntity.java:559]: You should add a newLine to match our code conventions.

File: src/main/java/salesforce/ui/pages/AppPageFactory.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package salesforce.ui.pages;
+
+import salesforce.ui.pages.legalentity.*;
+
```
Comment (Raymundogs) [inline src/main/java/salesforce/ui/pages/AppPageFactory.java:12]: You should delete this newLine.

File: src/main/java/salesforce/ui/pages/AppPageFactory.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import static core.config.LoadEnvironmentFile.getTheSalesforceVersion;
+
+public class AppPageFactory {
+
+    private static String version = getTheSalesforceVersion();
+
+    public static LegalEntitiesPageAbstract getLegalEntitiesPage() {
```
Comment (Raymundogs) [inline src/main/java/salesforce/ui/pages/AppPageFactory.java:24]: Missing javadoc.

File: src/main/java/salesforce/ui/pages/AppPageFactory.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        if ("classic".equals(version)) {
+            return new LegalEntitiesClassicVersion();
+        }
+        return new LegalEntitiesLightningVersion();
+    }
+
+    public static NewLegalEntityPageAbstract getNewLegalEntityPage() {
```
Comment (Raymundogs) [inline src/main/java/salesforce/ui/pages/AppPageFactory.java:36]: missing javadoc.

File: src/main/java/salesforce/ui/pages/AppPageFactory.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        if ("classic".equals(version)) {
+            return new NewLegalEntityClassicVersion();
+        }
+        return new NewLegalEntityLightningVersion();
+    }
+
+    public static LegalEntityPageAbstract getLegalEntityPage() {
```
Comment (Raymundogs) [inline src/main/java/salesforce/ui/pages/AppPageFactory.java:48]: missing javadoc.

File: src/main/java/salesforce/ui/pages/TableExtract.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+public class TableExtract {
+
+    private String tableHeaders = "div.slds-cell-fixed span.slds-truncate";
+    private String tableRow = "//a[@title='%s']/../../..//*[contains(@class,'slds-truncate')]";
+
+    /**
+     * Gets the table of the feature with the required element.
+     *
+     * @param map provided map
+     * @return Map<String, String>
```
Comment (Raymundogs) [inline src/main/java/salesforce/ui/pages/TableExtract.java:20]: You should provide a description of the returned map.

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java [hunk 7]
```diff
@@ -0,0 +1,160 @@
+package salesforce.ui.pages.legalentity;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
```
Comment (Raymundogs) [inline src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java]: You should delete newLine.

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+     *
+     * @return a String with address text.
+     */
+    public List<String> getAddressText() {
+        List<String> separatedAddress = new ArrayList<String>();
+        String theAddress = null;
+        if (fixAddress()) {
+            if (!getDriver().findElement(addressId).getText().isEmpty()) {
+                theAddress = getDriver().findElement(addressId).getText();
+                System.out.println("entro al if ray =>  " + theAddress);
```
Comment (Raymundogs) [inline src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java]: You should delete this println.

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+
+    /**
+     * Verify if a element is present.
+     *
+     * @return true if the element is present, false otherwise.
+     */
+    public boolean fixAddress() {
+        System.out.println();
+        if (getWebElementAction().isElementPresent(addressId, time)) {
+            System.out.println("elemento presente");
```
Comment (Raymundogs) [inline src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java]: You should delete this println.

File: src/main/java/salesforce/ui/pages/legalentity/NewLegalEntityClassicVersion.java [hunk 10]
```diff
@@ -0,0 +1,173 @@
+package salesforce.ui.pages.legalentity;
```
Comment (Raymundogs) [inline src/main/java/salesforce/ui/pages/legalentity/NewLegalEntityClassicVersion.java:9]: You should add copyright to match our teams conventions.

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityPageAbstract.java [hunk 11]
```diff
@@ -0,0 +1,18 @@
+package salesforce.ui.pages.legalentity;
```
Comment (Raymundogs) [inline src/main/java/salesforce/ui/pages/legalentity/LegalEntityPageAbstract.java:9]: You should add copyright to match our teams conventions.

File: src/test/java/cucumber/steps/LegalEntitySteps.java [hunk 12]
```diff
@@ -1,106 +1,78 @@
...
-    public void theCreatedLegalEntityShouldBeDisplayedOnTheLegalEntitiesTable() {
-        logger.info("=================== And The created legal entity should be on table ==========================");
-        SoftAssert softAssert = new SoftAssert();
-        LegalEntitiesPage newLegalEntitiesPage = new LegalEntitiesPage();
-        legalEntity.setId(newLegalEntitiesPage.getLegalEntityId(legalEntity.getName()));
-        softAssert.assertAll();
-    }
-
-}
+///**
```
Comment (Raymundogs) [inline src/test/java/cucumber/steps/LegalEntitySteps.java]: You should delete duplicated class.

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java (hunk 7 above)
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java]: delete blank line

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+     */
+    @Override
+    public String getHeaderEntityNameText() {
+        return fixText(getWebElementAction().getTextOnWebElement(headerName));
+    }
+
+    /**
+     * There is no message in the lightning skin.
+     *
+     * @return .
```
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java]: description for return in javadoc is missing

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java (hunk 8 above)
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java]: Delete print line

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+                theAddress = getDriver().findElement(addressId).getText();
+                System.out.println("entro al if ray =>  " + theAddress);
+                separatedAddress.addAll(Arrays.asList(theAddress.split("\n")));
+            } else {
+                separatedAddress.add(null);
+                separatedAddress.add(null);
+                separatedAddress.add(null);
+            }
+        }
+        System.out.println(separatedAddress);
```
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java]: Delete print lines. Check print lines in other methods of the class

File: src/main/java/salesforce/ui/pages/legalentity/NewLegalEntityLightningVersion.java [hunk 15]
```diff
@@ -1,28 +1,16 @@
...
 import core.selenium.WebElementAction;
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
-import org.openqa.selenium.support.ui.ExpectedConditions;
-import salesforce.entities.legalentity.LegalEntity;
-import salesforce.ui.pages.BasePage;
+import salesforce.entities.LegalEntity;
+import salesforce.ui.pages.AppPageFactory;
+
```
Comment (JessickaMoya) [inline src/main/java/salesforce/ui/pages/legalentity/NewLegalEntityLightningVersion.java]: Delete blank line

File: src/test/java/cucumber/steps/LegalEntitySteps.java (hunk 12 above)
Comment (JessickaMoya) [inline src/test/java/cucumber/steps/LegalEntitySteps.java]: Delete commented class

File: src/test/java/cucumber/steps/LegalEntitySteps2.java [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.testng.asserts.SoftAssert;
+import salesforce.entities.LegalEntity;
+import salesforce.ui.pages.AppPageFactory;
+import salesforce.ui.pages.legalentity.LegalEntityPageAbstract;
+import salesforce.ui.pages.legalentity.NewLegalEntityPageAbstract;
+import static salesforce.config.EnvironmentConfig.getSalesforceVersion;
+
+public class LegalEntitySteps2 {
```
Comment (JessickaMoya) [inline src/test/java/cucumber/steps/LegalEntitySteps2.java]: Change the class name to just "LegalEntitySteps"

File: src/main/java/salesforce/entities/LegalEntity.java (hunk 1 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/entities/LegalEntity.java:559]: ok :)

File: src/main/java/salesforce/ui/pages/AppPageFactory.java (hunk 2 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/ui/pages/AppPageFactory.java:12]: 👍🏻

File: src/main/java/salesforce/ui/pages/AppPageFactory.java (hunk 3 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/ui/pages/AppPageFactory.java:24]: 👍🏻

File: src/main/java/salesforce/ui/pages/AppPageFactory.java (hunk 4 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/ui/pages/AppPageFactory.java:36]: 👍🏻

File: src/main/java/salesforce/ui/pages/AppPageFactory.java (hunk 5 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/ui/pages/AppPageFactory.java:48]: 👍🏻

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java (hunk 7 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java]: ok :)

**Comment by JessickaMoya** [comment]
Everything is good. But all checks are failing

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java (hunk 8 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java]: okay, i'll correct it.

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java (hunk 9 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java]: okay, i'll correct it.

File: src/main/java/salesforce/ui/pages/legalentity/NewLegalEntityClassicVersion.java (hunk 10 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/ui/pages/legalentity/NewLegalEntityClassicVersion.java:9]: okay

File: src/test/java/cucumber/steps/LegalEntitySteps.java (hunk 12 above)
Comment (danielaSantaCruz) [reply src/test/java/cucumber/steps/LegalEntitySteps.java]: you're right i will delete it

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java (hunk 7 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java]: ok

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityPageAbstract.java (hunk 11 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/ui/pages/legalentity/LegalEntityPageAbstract.java:9]: okay

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java (hunk 13 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java]: 👍🏻

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java (hunk 8 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java]: already done

File: src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java (hunk 14 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/ui/pages/legalentity/LegalEntityClassicVersion.java]: already done.

File: src/test/java/cucumber/steps/LegalEntitySteps.java (hunk 12 above)
Comment (danielaSantaCruz) [reply src/test/java/cucumber/steps/LegalEntitySteps.java]: already done.

File: src/main/java/salesforce/ui/pages/legalentity/NewLegalEntityLightningVersion.java (hunk 15 above)
Comment (danielaSantaCruz) [reply src/main/java/salesforce/ui/pages/legalentity/NewLegalEntityLightningVersion.java]: ok

File: src/test/java/cucumber/steps/LegalEntitySteps2.java (hunk 16 above)
Comment (danielaSantaCruz) [reply src/test/java/cucumber/steps/LegalEntitySteps2.java]: already changed

**Review by Raymundogs — APPROVED** [review]

**Review by JessickaMoya — APPROVED** [review]

File: src/main/java/salesforce/ui/pages/AppPageFactory.java [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+        return new NewLegalEntityLightningVersion();
+    }
+
+    /**
+     * Gets the legal entity page.
+     *
+     * @return the correct legal entity page instance.
+     */
+    public static LegalEntityPageAbstract getLegalEntityPage() {
+        if ("classic".equals(version)) {
```
Comment (Paola-Xaq) [inline src/main/java/salesforce/ui/pages/AppPageFactory.java:49]: I think comparing must have to the right version.equals("classic")

**Review by Paola-Xaq — APPROVED** [review]

**Pull request merged**